Tribler/Core/Upgrade/pickle_converter.py

```diff
@@ -1,149 +1,82 @@
-import io
-import os
+import cPickle
 import glob
-import json
+import os
 import pickle
-import cPickle
-import StringIO
-
 from ConfigParser import RawConfigParser
 
-from Tribler.Core.SessionConfig import SessionStartupConfig, SessionConfigInterface
-from Tribler.Core.DownloadConfig import DefaultDownloadStartupConfig, get_default_dscfg_filename
-from Tribler.Core.simpledefs import PERSISTENTSTATE_CURRENTVERSION, STATEDIR_GUICONFIG
+from Tribler.Core.simpledefs import PERSISTENTSTATE_CURRENTVERSION
 
 
 class PickleConverter(object):
     """
-    This class is responsible for converting old .pickle files used for configuration files to a newer .state format.
+    This class is responsible for converting old .pickle files used for configuration files to a newer ConfigObj format.
     """
 
     def __init__(self, session):
         self.session = session
 
     def convert(self):
         """
-        Calling this method will convert all configuration files to the newer .state format.
+        Calling this method will convert all configuration files to the ConfigObj/.state format.
         """
         self.convert_session_config()
         self.convert_main_config()
-        self.convert_default_download_config()
         self.convert_download_checkpoints()
 
     def convert_session_config(self):
         """
-        Convert the sessionconfig.pickle file to libtribler.conf. Do nothing if we do not have a pickle file.
+        Convert the sessionconfig.pickle file to triblerd.conf. Do nothing if we do not have a pickle file.
+        Remove the pickle file after we are done.
         """
-        old_filename = os.path.join(self.session.get_state_dir(), 'sessconfig.pickle')
-        new_filename = SessionConfigInterface.get_default_config_filename(self.session.get_state_dir())
+        old_filename = os.path.join(self.session.config.get_state_dir(), 'sessconfig.pickle')
 
         if not os.path.exists(old_filename):
             return
 
         with open(old_filename, "rb") as old_file:
             sessconfig = pickle.load(old_file)
 
-        # Upgrade to new config
-        sconfig = SessionStartupConfig()
+        # Upgrade to .state config
+        new_config = self.session.config
```

Comment:
In this case, the `SessionStartupConfig` class is removed. What would you do in this case?